tezos/api/src/ocaml_conv/from_ocaml.rs

```diff
@@ -32,22 +34,44 @@ macro_rules! from_ocaml_hash {
     };
 }
 
+macro_rules! from_ocaml_typed_hash {
+    ($ocaml_name:ident, $rust_name:ident) => {
+        unsafe impl FromOCaml<$ocaml_name> for $rust_name {
+            fn from_ocaml(v: OCaml<$ocaml_name>) -> Self {
+                unsafe {
+                    let vec: Vec<u8> = v.field::<OCamlBytes>(0).to_rust();
+                    use std::convert::TryFrom;
+                    $rust_name::try_from(vec).unwrap()
```

Comment:
Is any processing required here or is this something that can be mapped from the OCaml bytes? (we can assume they are correct I suppose)

Reply:
Is there any way we can receive bytes of wrong length from OCaml? Do you think we should state that clearer with `unreachable!()`?

Reply:
No, what I meant is: `try_from` has to do some computation to validate those bytes. But we know that these values that come from the OCaml side are valid already, right? So knowing that, cannot we just map the bytes directly and avoid the cost of performing the validation?

Reply:
@akoptelov sorry, I confused what happens in `try_from` with what happens inside `b58check_to_hash`. In this case it just checks the length, so it is fine. But adding that `unreachable!()` as you did is good, thanks.